.github/workflows/ci.yml

```diff
@@ -31,18 +41,27 @@ jobs:
     - name: Binary compatibility settings (Windows)
       if: ${{ startsWith(matrix.os, 'windows') }}
       run: echo 'RUSTFLAGS=-Ctarget-feature=+crt-static' >> $GITHUB_ENV
-    # Run gradle normally to build the JNI lib
-    - name: Build with Gradle
-      if: ${{ !startsWith(matrix.os, 'ubuntu') }}
-      run: ./gradlew build copyJniLib
+    - name: Set build target for cross-compiling
+      if: matrix.target != ''
+      run: |
+        echo CARGO_BUILD_TARGET=${{ matrix.target }} >> $GITHUB_ENV
+        echo JNILIB_RUST_TARGET=${{ matrix.target }} >> $GITHUB_ENV
+        echo GRADLE_ARGS="${GRADLE_ARGS} -x test">> $GITHUB_ENV
+        rustup target add ${{ matrix.target }}
     # Run gradle in a docker container to build the JNI lib on Linux for old glibc compatibility
     - name: Build with Gradle (Linux)
       if: ${{ startsWith(matrix.os, 'ubuntu') }}
       run: |
         docker build -t build-image ./ci/docker/x86_64-linux
         docker run --rm --volume $PWD:/build --workdir /build build-image ./gradlew build copyJniLib -x javadoc
+    # Otherwise, run gradle normally to build the JNI lib
+    - name: Build with Gradle
+      if: ${{ !startsWith(matrix.os, 'ubuntu') }}
+      run: ./gradlew build copyJniLib ${GRADLE_ARGS}
+    - name: List shared library files
+      run:
+        ls build/jni-libs
     - name: Save JNI lib output
-      if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Is this change intended? This will attempt to upload the built jnilibs for non-tagged commits.